src/features/list-view/ListView.js

```diff
@@ -67,8 +64,51 @@ class ListView extends React.PureComponent<Props> {
         );
     };
 
+    /**
+     * Returns the width of the target column using one of three possible scenarios:
+     * 1. Delegate to props.getColumnWidth
+     * 2. Default to DEFAULT_COLUMN_WIDTH
+     * 3. Stretch to fit the browser viewport
+     *
+     * Column-stretching occurs when the total columns combined is shorter than width of MultiGrid.
+     * The left-most column is excluded from stretching.
+     *
+     * TODO: Force widths to be recomputed when container width changes.
+     */
+    computeColumnWidth = ({ index }: ComputeColumnWidthArgs) => {
+        const columnIndex = index;
+        const { columnCount, getColumnWidth, width } = this.props;
+
+        // Returns the client-defined width or a default value.
+        const getClientDefinedColumnWidth = columnIndex_ =>
+            getColumnWidth ? getColumnWidth(columnIndex_) : DEFAULT_COLUMN_WIDTH;
+
+        if (columnCount === 0) {
+            return 0;
+        }
+
+        const firstColumnWidth = getClientDefinedColumnWidth(0);
+        if (columnIndex === 0) {
+            return firstColumnWidth;
+        }
+
+        // Determine if columns fill width of
+        // MultiGrid or stretching is needed.
+        let canFillWidth = false;
```

Comment:
nitpick: can we move var declarations to the top of the function?

Reply:
Can you help me understand why you prefer that? 

Moving it to the top means any reader must jump their eyeballs up and down the function to make sense of it.

Reply:
Thats true, i've always followed the pattern of putting all var declarations at the top of the function. That way they're in one place and you always know where to look.